Foundation/Collection/Foldable.hs

```diff
@@ -43,6 +45,20 @@ class Foldable collection where
     foldr' :: (Element collection -> a -> a) -> a -> collection -> a
     foldr' f z0 xs = foldl f' id xs z0 where f' k x z = k $! f x z
 
+-- | Fold1's. Like folds, but they assume to operate on a NonEmpty collection.
+class Foldable f => Fold1able f where
+    -- | Left associative lazy fold.
+    foldl1  :: (Element f -> Element f -> Element f) -> NonEmpty f -> Element f
```

Comment:
indeed. Still, might be worth removing it. I don't believe anyone would use the lazy version of `foldl` or `foldl1` properly anyway. There is a big warning on `base`'s `foldl`... some (including me) believe we should better remove it.